plugin/src/software/aws/toolkits/eclipse/amazonq/handlers/QOpenLoginViewHandler.java

```diff
@@ -5,18 +5,53 @@
 
 import org.eclipse.core.commands.AbstractHandler;
 import org.eclipse.core.commands.ExecutionEvent;
-import software.aws.toolkits.eclipse.amazonq.plugin.Activator;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.IViewReference;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
 
+import software.aws.toolkits.eclipse.amazonq.plugin.Activator;
+import software.aws.toolkits.eclipse.amazonq.views.AmazonQView;
 import software.aws.toolkits.eclipse.amazonq.views.ViewVisibilityManager;
 
 public class QOpenLoginViewHandler extends AbstractHandler {
+
     @Override
     public final Object execute(final ExecutionEvent event) {
+        closeAnyOpenQViews();
```

Comment:
this doesn't seem necessary. The showChatView call already does the job hiding any open views that are not the view being asked to be shown

Reply:
Do you mean the mutually exclusive views? It doesn't include the AmazonQView though. It just handles the views inside of the base view. The base view itself remains open.

Reply:
Did not get it. Each of the two classes than implement the baseview are part of the mutually exclusive view.
Why would this need to handled separately from the mutuallyexclusive call instead?

Reply:
Sorry, I should've picked a better name. But basically, the AmazonQView is the skeleton, and all the other views: ChatWebview, DependencyMissingView, and LoginView are all rendered inside it. These views are mutually exclusive, but the skeleton itself is not part of that process. It can remain open without any contents inside it, which is what happens when the window is switched. This logic finds all instances of the skeleton view in any of the open windows and closes them before opening a new skeleton view in some active shell.

Reply:
This logic is missing should be moved to within the showMutuallyExclusive call instead of adding it to a third separate location to manage and maintain long term